src/hooks/useScroll.ts

```diff
@@ -0,0 +1,20 @@
+import type { ScrollRef } from '../types';
+
+export const useScroll = () => {
```

Comment:
Really nice example when something should not be hook. Ideally, hooks is the thing which uses other hooks like useContext, useState or creates handlers using state / props from the component.
```suggestion
export const useScroll = () => {
```